pkg/conn/basedb_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conn
+
+import (
+	"github.com/DATA-DOG/go-sqlmock"
+	. "github.com/pingcap/check"
+	tcontext "github.com/pingcap/dm/pkg/context"
+	"github.com/pingcap/dm/pkg/retry"
+)
+
+var _ = Suite(&testBaseDBSuite{})
+
+type testBaseDBSuite struct {
+}
+
+func (t *testBaseDBSuite) TestBaseDB(c *C) {
+	db, mock, err := sqlmock.New()
+	c.Assert(err, IsNil)
+
+	baseDB := &BaseDB{db, &retry.FiniteRetryStrategy{}}
+
+	tctx := tcontext.Background()
+
+	dbConn, err := baseDB.GetBaseConn(tctx.Context())
+	c.Assert(dbConn, NotNil)
+	c.Assert(err, IsNil)
+
+	mock.ExpectQuery("select 1").WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow("1"))
+	rows, err := dbConn.QuerySQL(tctx, "select 1")
+	c.Assert(err, IsNil)
+	ids := make([]int, 0, 1)
+	for rows.Next() {
+		var id int
+		err = rows.Scan(&id)
+		c.Assert(err, IsNil)
+		ids = append(ids, id)
+	}
+	c.Assert(len(ids), Equals, 1)
```

Comment:
```suggestion
	c.Assert(ids, HasLen, 1)
```

Reply:
Done in [here](https://github.com/pingcap/dm/pull/266/files#diff-8dd3ef1da219ede5563625a2fc6ceb7cR51)